cpp/src/arrow/flight/client.cc

```diff
@@ -198,6 +198,8 @@ class FlightClient::FlightClientImpl {
     grpc::ChannelArguments args;
     // Try to reconnect quickly at first, in case the server is still starting up
     args.SetInt(GRPC_ARG_INITIAL_RECONNECT_BACKOFF_MS, 100);
+    // Receive messages of any size
+    args.SetMaxReceiveMessageSize(-1);
```

Comment:
Note the issue in https://nanxiao.me/en/message-length-setting-in-grpc/ was fixed in https://github.com/grpc/grpc/issues/13553